### ISSUE TITLE
Normalization and BatchNormalization layer does not rescale or normalize inputs, am I missing anything?

### ISSUE THREAD

**Issue by jayagami** [tensorflow/tensorflow#55343]
**System information**

colab with tf-2.8
[old-tf_env.txt](https://github.com/tensorflow/tensorflow/files/8330439/old-tf_env.txt)

---

**Describe the current behavior**

Normalization and Batch Normalization layer does not rescale or normalize inputs as excepted, the inputs barely not changing.

---

**Standalone code to reproduce the issue**

###  Normalization layer
```python
tfdata = np.array([[1,2,3,4,5]],dtype=float)
print("tfdata:" ,tfdata)
print("avg: ", np.average(tfdata))
print("var: ", np.var(tfdata))
```
>tfdata: [[1. 2. 3. 4. 5.]]
avg:  3.0
var:  2.0

```python
normalization_layer = tf.keras.layers.Normalization(axis=-1)
normalized_data = normalization_layer(tfdata)
print(normalized_data.numpy())
print("normalized avg: ", np.average(normalized_data.numpy()))
print("normalized var: ", np.var(normalized_data.numpy()))
```
**output:** 

>[[1. 2. 3. 4. 5.]]
normalized avg:  3.0
normalized var:  2.0

**This output is unchanged**

### BatchNormalization layer

```python

batch_normalization_layer = tf.keras.layers.BatchNormalization(axis=-1,scale=True,center=True)
batch_normalized_data = batch_normalization_layer(tfdata)
print(batch_normalized_data.numpy())
print("batch normalized avg: ", np.average(batch_normalized_data.numpy()))
print("batch normalized var: ", np.var(batch_normalized_data.numpy()))
```
**output:**
>[[0.9995004 1.9990008 2.9985013 3.9980016 4.997502 ]]
batch normalized avg:  2.9985013
batch normalized var:  1.9980018

---

**This output still unchanged!**

### LayerNorm

```python
tdata = torch.tensor([1,2,3,4,5],dtype=torch.float32)
normalization_layer=torch.nn.LayerNorm(normalized_shape=[5])
normalized_data = normalization_layer(tdata).detach().numpy()
print(normalized_data)
print("normalized avg: ", np.average(normalized_data))
print("normalized var: ", np.var(normalized_data))
```

**output**

>[-1.4142101  -0.70710504 -0.00000006  0.7071049   1.4142098 ]
normalized avg:  -4.7683717e-08
normalized var:  0.9999949

normalized av closed to 0, and normalized var close to 1, correct answer.

and BatchNormalization ... all the same

**Comment by tilakrayal** [comment]
@jayagami ,
Can you please elaborate about your issue here.Also please confirm which output you are mentioning as the right one in both? It helps to debug the issue.Thanks!

**Comment by jayagami** [comment]
> @jayagami , Can you please elaborate about your issue here.Also please confirm which output you are mentioning as the right one in both? It helps to debug the issue.Thanks!

Thanks for replying.

Well,  both Normalization and BatchNormalization layer in tf are not working.

according to BatchNormalization definition:

![image](https://user-images.githubusercontent.com/40586590/159834449-b71459b6-b513-4fec-bf9e-fa34742744dc.png)


In the outputs, the mean should be close to 0, and the variance should be close to 1. (ignore the gamma, beta……)
However, as you can see, the outputs are unchanged.



As a comparison, Torch gives the expected answer:

>normalized avg: -4.7683717e-08
normalized var: 0.9999949

By the way, Layer Normalization is working as excepted in tf 2.8.

This is a very basic layer, and I doubt I'm missing something.

**Comment by jayagami** [comment]
Also, I tried to set mean and variance

mean=0, variance=1.0

```python
input_data = np.array([[1.], [2.], [3.]], dtype='float32')
layer = tf.keras.layers.Normalization(mean=0., variance=1.)
layer(input_data)
```
the output is 

><tf.Tensor: shape=(3, 1), dtype=float32, numpy=
array([[1.],
       [2.],
       [3.]], dtype=float32)>


and
mean=100
variance = 1

```python

input_data = np.array([[1.], [2.], [3.]], dtype='float32')
layer = tf.keras.layers.Normalization(mean=100., variance=0.1.)
layer(input_data)

```

output:

><tf.Tensor: shape=(3, 1), dtype=float32, numpy=
array([[-99.],
       [-98.],
       [-97.]], dtype=float32)>


besides, this output is different with examples in  

```python
adapt_data = np.array([[0., 7., 4.],
                       [2., 9., 6.],
                       [0., 7., 4.],
                       [2., 9., 6.]], dtype='float32')
input_data = np.array([[0., 7., 4.]], dtype='float32')
layer = tf.keras.layers.Normalization(axis=-1)
layer.adapt(adapt_data)
layer(input_data)
```

outputs:

><tf.Tensor: shape=(1, 3), dtype=float32, numpy=array([[-1., -1., -1.]], dtype=float32)>



I am really confused.

**Comment by jayagami** [comment]
I checked the code, could be a keras problem.


The call method is quite simple, seems it did not calculate mean and variance from input data.

https://github.com/keras-team/keras/blob/v2.8.0/keras/layers/preprocessing/normalization.py#L300-L306

```python
  def call(self, inputs):
    inputs = self._standardize_inputs(inputs)
    # The base layer automatically casts floating-point inputs, but we
    # explicitly cast here to also allow integer inputs to be passed
    inputs = tf.cast(inputs, self.compute_dtype)
    return ((inputs - self.mean) /
            tf.maximum(tf.sqrt(self.variance), backend.epsilon()))
```

and the mean is initialized as 0, variance is initialized as 1, unless you manually set it.

https://github.com/keras-team/keras/blob/v2.8.0/keras/layers/preprocessing/normalization.py#L244-L298

**Comment by jayagami** [comment]
I re-checked the documentation and found it right in front of me, surprised I hadn't noticed this paragraph before.
I understand the need for this design, which is very different from torch. I should have paid more attention to these details.
